src/compiler/transformers/classFields.ts

```diff
@@ -1802,99 +1976,61 @@ namespace ts {
-                        isValid,
-                    });
-                }
-                else if (isMethodDeclaration(node)) {
-                    const functionName = createHoistedVariableForPrivateName(text, node);
-                    privateEnv.identifiers.set(privateName, {
-                        kind: PrivateIdentifierKind.Method,
-                        methodName: functionName,
-                        brandCheckIdentifier: classConstructor,
-                        isStatic: true,
-                        isValid,
-                    });
-                }
-                else if (isGetAccessorDeclaration(node)) {
-                    const getterName = createHoistedVariableForPrivateName(text + "_get", node);
-                    if (previousInfo?.kind === PrivateIdentifierKind.Accessor && previousInfo.isStatic && !previousInfo.getterName) {
-                        previousInfo.getterName = getterName;
-                    }
-                    else {
-                        privateEnv.identifiers.set(privateName, {
-                            kind: PrivateIdentifierKind.Accessor,
-                            getterName,
-                            setterName: undefined,
-                            brandCheckIdentifier: classConstructor,
-                            isStatic: true,
-                            isValid,
-                        });
-                    }
-                }
-                else if (isSetAccessorDeclaration(node)) {
-                    const setterName = createHoistedVariableForPrivateName(text + "_set", node);
-                    if (previousInfo?.kind === PrivateIdentifierKind.Accessor && previousInfo.isStatic && !previousInfo.setterName) {
-                        previousInfo.setterName = setterName;
-                    }
-                    else {
-                        privateEnv.identifiers.set(privateName, {
-                            kind: PrivateIdentifierKind.Accessor,
-                            getterName: undefined,
-                            setterName,
-                            brandCheckIdentifier: classConstructor,
-                            isStatic: true,
-                            isValid,
-                        });
-                    }
-                }
-                else {
-                    Debug.assertNever(node, "Unknown class element type.");
-                }
+        function addPrivateIdentifierClassElementToEnvironment<T extends PropertyDeclaration | MethodDeclaration | GetAccessorDeclaration | SetAccessorDeclaration>(node: T, name: PrivateIdentifier, lex: ClassLexicalEnvironment, privateEnv: PrivateIdentifierEnvironment, isStatic: boolean, isValid: boolean, previousInfo: PrivateIdentifierInfo | undefined) {
+            if (isAutoAccessorPropertyDeclaration(node)) {
+                addPrivateIdentifierAutoAccessorPropertyDeclarationToEnvironment(node, name, lex, privateEnv, isStatic, isValid, previousInfo);

```

Comment:
That ends up being far less type safe since returning the function based on a narrower type doesn't narrow the type itself, so you have to have the result accept a wider type.